airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -96,15 +95,9 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
 
             api = API(account_id=config.account_id, access_token=config.access_token, page_size=config.page_size)
 
-            record_iterator = AdAccount(api=api).read_records(sync_mode=SyncMode.full_refresh, stream_state={})
-            account_info = list(record_iterator)[0]
-
-            if account_info.get("is_personal"):
-                message = (
-                    "The personal ad account you're currently using is not eligible "
-                    "for this operation. Please switch to a business ad account."
-                )
-                return False, message
+            # Get Ad Account to check creds
+            ad_account = api.account
```

Comment:
isn't api.account a lazy attribute? if it is, let's add a small unit test for this change because if we'll somehow drop the log line, we can miss connectivity check at all

Reply:
Based on our tests and the documentation, the lazy property is indeed evaluated on first access. I've added a test to ensure that credentials validation is consistently checked.

[cached_property docs](https://docs.python.org/3/library/functools.html#functools.cached_property):
> The cached_property decorator only runs on lookups and only when an attribute of the same name doesn’t exist. When it does run, the cached_property writes to the attribute with the same name. Subsequent attribute reads and writes take precedence over the cached_property method and it works like a normal attribute.